libraries/Microsoft.Bot.Builder/HealthCheck.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder
+{
+    internal class HealthCheck
+    {
+        public static HealthCheckResponse CreateHealthCheckResponse(IConnectorClient connector)
```

Comment:
I think I'm missing the point of this. I had expected the Health Check to be a simple Invoke, and the just pretty much just responds as "Ok". From the spec, that seems to be what should be happening. 

I'm missing the understanding on the headers and related bits your capturing.

Reply:
The spec says return the Authorization header that would be used on a callback and the UserAgent that would be added to that callback. So its going the extra mile on validation. I've tested end to end all the scenarios (in skills which is all I can do) there is many - for example expectReplies doesn't actually need to do a callback to work.

This is override able by a custom healthcheck. For example, a derived class could add messages to the message array.

So I'm completely faithful to that - I actually invoke the connector as if the callback was being made.